packages/core/src/component.ts

```diff
@@ -3,34 +3,49 @@
 // This file is licensed under the MIT License.
 // License text available at https://opensource.org/licenses/MIT
 
-import {Constructor, Provider, BoundValue} from '@loopback/context';
-import {Application} from '.';
-
-// tslint:disable:no-any
+import { Constructor, Provider, BoundValue, Context } from '@loopback/context';
+import {Server} from '.';
 
 export interface ProviderMap {
   [key: string]: Constructor<Provider<BoundValue>>;
 }
 
 export interface Component {
-  controllers?: Constructor<any>[];
+  controllers?: Array<Constructor<{}>>;
   providers?: ProviderMap;
+
+  // tslint:disable-next-line:no-any
   [prop: string]: any;
 }
 
+/**
+ * Mount a component to a Context. Examples can include mounting components
+ * to Server instances to attach server-specific components (ex. adding
+ * authentication to a RestServer instance).
+ *
+ * Components can also be mounted to Application instances to provide app-wide
+ * access to components. (ex. adding a messaging service to allow a RestServer
+ * to communicate with another server, like gRPC)
+ *
+ * @export
+ * @param {Context} context
+ * @param {Component} component
+ */
 export function mountComponent(
-  app: Application,
+  context: Context,
   component: Component,
 ) {
+  // TODO: Incorporate namespacing into this?
   if (component.controllers) {
-    for (const controllerCtor of component.controllers) {
-      app.controller(controllerCtor);
```

Comment:
I am not very happy about this change.

First of all, not every context is a good target for a component - e.g. I don't think mounting a component on the base `Context` instance is a valid usage.

Secondly, what if the user has a custom `Application` class that has overridden `app.controller` method? Before your change, `mountComponent` would call this custom method. After the change, the custom method is not called - I would personally find that confusing as a user.

Thoughts?